### ISSUE TITLE
False positive 'Unnecessary semicolon' and 'Missing trailing comma before ";"' for enums with companion object

### ISSUE THREAD

**Issue by oleksandr-bezushko-axon** [pinterest/ktlint#1743]
```kotlin
package some.package

enum class TestEnum(val someField: String) {
    A("a"),
    B("b"),
    C("c");

    companion object {
        fun findBySomeField(field: String): TestEnum? = values().find { it.someField == field }
    }
}
```

If run this snipped of code with ktlint v0.48.0, you'll get the following output:

```
/path/to/TestEnum.kt:6:11: Unnecessary semicolon (no-semi)
/path/to/TestEnum.kt:6:11: Missing trailing comma before ";" (trailing-comma-on-declaration-site)

Summary error count (descending) by rule:
  no-semi: 1
  trailing-comma-on-declaration-site: 1
```

Telling ktlint to resolve the issues by running `ktlint -F /path/to/TestEnum.kt` produces the following code:

```kotlin
package some.package

enum class TestEnum(val someField: String) {
    A("a"),
    B("b"),
    C("c")

    companion object {
        fun findBySomeField(field: String): TestEnum? = values().find { it.someField == field }
    },
}
```

which is an invalid code:

![image](https://user-images.githubusercontent.com/75365247/209341048-f0268bd2-3425-4e89-b8fe-6f4b1dd3d3cb.png)

![image](https://user-images.githubusercontent.com/75365247/209341080-b2b5d02e-3245-49c6-94d2-50392a710b77.png)

## Your Environment

* Version of ktlint used: 0.48.0
* Relevant parts of the `.editorconfig` settings: not used
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): ktlint CLI v0.48.0
* Version of Gradle used (if applicable): not applicable
* Operating System and version: archlinux

**Comment by paul-dingemans** [comment]
Duplicate of #1733
